### ISSUE TITLE
WEBDAV publically accessible despite enviroment settings

### ISSUE THREAD

**Issue by kyle-mckay** [loganmarchione/docker-webdav-nginx#18]
Hopefully I'm just being stupid here and missing something, as your solution seems fairly strait forward and simple. I spun up a compose with user, password and file size all defined, only different thing was the volume location mapping to the host that was changed. The issue, is that when I connect to the webdav to test through the browser there is no prompt for authentication and if anyone found out about the URL they could access the files. 

![2023-08-25_14h39_46](https://github.com/loganmarchione/docker-webdav-nginx/assets/83886427/81115f5b-b41d-4eb2-8fb6-80a09ff8d721)

For context if it matters, the host port is being passed through NGINX proxy manager.

my docker compose
``` YML
version: '3'
services:
  webdav:
    container_name: docker-webdav-nginx
    restart: unless-stopped
    environment:
      - WEBDAV_USER=${USER}
      - WEBDAV_PASS=${PASSWORD}
      - NGINX_CLIENT_MAX_BODY_SIZE=${MAX_FILE_SIZE} #ex 500M, 0 for no limit
    networks:
      - webdav
    ports:
      - '8617:80'
    volumes:
      - '/e/userfiles/users/exampleuser:/var/www/webdav'
    image: loganmarchione/docker-webdav-nginx:latest

networks:
  webdav:

#volumes:
#  webdav:
#    driver: local
```

**Comment by loganmarchione** [comment]
What path are you visiting?

/public - No authentication
/restricted - Authentication if specified

**Comment by kyle-mckay** [comment]
Neither. The only path being used is as per your example compose which is `/var/www/webdav`. Based on the Explanation section in the readme I honestly assumed it would dynamically assign the path if authentication variables were defined. Would I just need to update the volume to `/var/www/webdav/restricted`?

**Comment by loganmarchione** [comment]
I mean in the browser?

**Comment by kyle-mckay** [comment]
I was simply putting in https://exampledav.domain.com or equivelent to `localhost:8617`. Using either ./public or ./restricted in the browser path result in a 404.
![image](https://github.com/loganmarchione/docker-webdav-nginx/assets/83886427/f82a92e4-3c0c-44c7-97bb-601e4c3b1c3c)

**Comment by kyle-mckay** [comment]
Saying that, I updated the container path to `/var/www/webdav/restricted` and spun the container back up and now I see the two options and it is prompting for authentication. So by having `${hostpath};/var/www/webdav` it is acting like the public and restricted entry points were overwritten.
![image](https://github.com/loganmarchione/docker-webdav-nginx/assets/83886427/217b0605-bed2-408a-9aaf-5fb36f0b2a25)
Example of the working docker-compose
```YML
version: '3'
services:
  webdav:
    container_name: docker-webdav-nginx
    restart: unless-stopped
    environment:
      - WEBDAV_USER=${USER}
      - WEBDAV_PASS=${PASSWORD}
      - NGINX_CLIENT_MAX_BODY_SIZE=${MAX_FILE_SIZE} #ex 500M or 0 for no limit
    networks:
      - webdav
    ports:
      - '8617:80'
    volumes:
      #- '/e/userfiles/users/kylemckay:/var/www/webdav' #public for some reason
      - '/e/userfiles/users/kylemckay:/var/www/webdav/restricted' #working with authentication
    image: loganmarchione/docker-webdav-nginx:latest

networks:
  webdav:

#volumes:
#  webdav:
#    driver: local
```

**Comment by loganmarchione** [comment]
Hmmm you shouldn't have to mount restricted into the container. 

My setup is here with just /var/www/webdav. 

https://github.com/loganmarchione/k8s_homelab/blob/master/cluster/apps/webdav/webdav.yaml#L51

**Comment by kyle-mckay** [comment]
does the host file structure on your end already contain public and restricted folders?

**Comment by loganmarchione** [comment]
I'm AFK right now but it should be created by the container. 

https://github.com/loganmarchione/docker-webdav-nginx/blob/master/Dockerfile#L19

**Comment by kyle-mckay** [comment]
It might be related to the bind mounting as you said. Is it possible that when it "creates" the directories, it is happening BEFORE the volume is defined and mapped like in [Line 26](https://github.com/loganmarchione/docker-webdav-nginx/blob/master/Dockerfile#L26C11-L26C11)? Once it is mapped to the host perhaps it is mirroring the host now instead of using the folders already created.

I'm not familiar with docker file creation, but maybe the mkdir's need to be after the volume is mapped? Like this instead? This could then CREATE the folders after being mapped to the host. 

```dockerfile
RUN apt-get update && apt-get -y install --no-install-recommends \
    apache2-utils \
    netcat-openbsd \
    nginx-extras && \
    rm -rf /var/lib/apt/lists/* && \
    mkdir  "/var/www/webdav" &&\
    rm /etc/nginx/sites-enabled/default

EXPOSE 80

VOLUME [ "/var/www/webdav" ]

RUN mkdir -p "/var/www/webdav/restricted" && \
    mkdir -p "/var/www/webdav/public" && \
    chown -R www-data:www-data "/var/www"
```

That being said, if their file structure is like mine they would still need to then move the contents into /restricted after the container is live in order to utilize authentication. At that point it seems easier to just say "If your host directory doesnt start with /public and /restricted, set the container directory to `./webdav/restricted` if you want authentication.

**Comment by loganmarchione** [comment]
Just spun up an instance using my existing [development compose file](https://github.com/loganmarchione/docker-webdav-nginx/blob/master/docker-compose-dev.yml) with a new volume (not a host bind mount) and it works ok for me (no need to mount `/restricted` or change the Dockerfile). I think any issues you had were due to the host mount.

It created the correct directories in the volume.

```
/var/lib/docker/volumes/docker-webdav-nginx_webdav2/_data/
├── public
└── restricted
```

Accessing the main page (without `/public` or `/restricted`) gives you access to both of those subpages (as it's intended to).
![Screenshot_20230827_122558](https://github.com/loganmarchione/docker-webdav-nginx/assets/4955337/306f09bd-b159-4559-8039-3edd7e86edc1)

That being said, I tried to [update the README](https://github.com/loganmarchione/docker-webdav-nginx/commit/917363386f66be81bd744e6e5675c90667346eb5) to be more clear. Thanks for the feedback!
